freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/upgrade/ccm/handler/UpgradeCcmChangeTunnelHandler.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.freeipa.flow.stack.upgrade.ccm.handler;
+
+import static com.sequenceiq.freeipa.flow.stack.upgrade.ccm.selector.UpgradeCcmHandlerSelector.UPGRADE_CCM_CHANGE_TUNNEL_EVENT;
+import static com.sequenceiq.freeipa.flow.stack.upgrade.ccm.selector.UpgradeCcmStateSelector.UPGRADE_CCM_FAILED_EVENT;
+import static com.sequenceiq.freeipa.flow.stack.upgrade.ccm.selector.UpgradeCcmStateSelector.UPGRADE_CCM_TUNNEL_CHANGE_FINISHED_EVENT;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+import com.sequenceiq.freeipa.flow.stack.upgrade.ccm.UpgradeCcmService;
+import com.sequenceiq.freeipa.flow.stack.upgrade.ccm.event.UpgradeCcmEvent;
+import com.sequenceiq.freeipa.flow.stack.upgrade.ccm.event.UpgradeCcmFailureEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class UpgradeCcmChangeTunnelHandler extends AbstractUpgradeCcmEventHandler {
```

Comment:
are you planning to add unit tests to the handler later?

Reply:
yes.
Currently it has an integration test implemented as a unit test